src/electron/helpers/services/comfyui.ts

```diff
@@ -0,0 +1,267 @@
+	 *
+	 * @param {number} timeout - The initial timeout for retrying the readiness check.
+	 * @param {number} maxRetries - The maximum number of retries for checking server readiness.
+	 * @returns {Promise<boolean>} True if the server is ready, false otherwise.
+	 */
+	private async checkServerReady(
+		timeout: number = 2000,
+		maxRetries: number = 60
+	): Promise<boolean> {
+		let retries = 0;
+		while (retries < maxRetries) {
+			try {
+				const response = await axios.get<SystemStats>(`http://${this.url}/system_stats`);
+				if (response.status === 200) {
+					return true;
+				}
+			} catch {
+				retries++;
+				await new Promise(resolve => {
+					setTimeout(resolve, timeout);
+				});
+			}
+		}
+
+		logger.error(`ComfyUI is not ready after ${maxRetries} retries`);
+
+		return false;
+	}
+
+	/**
+	 * Sets up a WebSocket connection for a given client ID and attaches message handling.
+	 * It manages the lifecycle events of the WebSocket connection including opening, messages, closing, and errors.
+	 *
+	 * @param {string} clientId - The unique identifier for the client.
+	 * @param {WebSocketMessageHandler} handleMessage - The function to call when a message is received on the WebSocket.
+	 * @private
+	 */
+	private setupWebSocket(clientId: string, handleMessage: WebSocketMessageHandler) {
+		const ws = new WebSocket(`ws://${this.url}/ws?clientId=${clientId}`);
+
+		ws.on("open", () => {
+			logger.info("WebSocket connection established", clientId);
+		});
+
+		ws.on("message", data => {
+			handleMessage(JSON.parse(data.toString()));
+		});
+
+		ws.on("close", () => {
+			logger.info("WebSocket connection closed", clientId);
+		});
+
+		ws.on("error", error => {
+			logError(error, `WebSocket error for ${clientId}`);
+		});
+
+		this.clients.set(clientId, ws);
+	}
+
+	/**
+	 * Registers a new client for WebSocket communications, setting up a new WebSocket connection
+	 * if one does not already exist for the given client ID.
+	 *
+	 * @param {string} clientId - The unique identifier for the client.
+	 * @param {WebSocketMessageHandler} handleMessage - The function to handle incoming WebSocket messages.
+	 */
+	public registerClient(clientId: string, handleMessage: WebSocketMessageHandler) {
+		if (this.clients.has(clientId)) {
+			logger.info(`Client ${clientId} is already registered.`);
+		} else {
+			this.setupWebSocket(clientId, handleMessage);
+		}
+	}
+
+	/**
+	 * Queues a new workflow for processing by the ComfyUI server.
+	 *
+	 * @param {object} prompt - The workflow data to be processed.
+	 * @param {string} clientId - The client ID associated with this request.
+	 * @returns {Promise<QueueWorkflowResponse>} The response from the server including prompt ID and any node errors.
+	 * @throws {Error} Throws an error if the request to queue the workflow fails.
+	 */
+	public async queueWorkflow(prompt: object, clientId: string): Promise<QueueWorkflowResponse> {
```

Comment:
we could use either `Record<string, unknown>` or the type we use in Broom

```ts
interface Node<T extends Record<string, unknown> = Record<string, unknown>> {
	inputs: T;
	class_type: string;
}


// Represents the node chain as a mapping from string to Node
export type NodeChain = Record<string, Node>;
```

Reply:
Added the type as suggested.